src/app/components/Routes/index.js

```diff
@@ -2,17 +2,27 @@ import React, { lazy } from 'react';
 import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
 
 import ROUTES from '~constants/routes';
+import { loadStorage } from '~utils/storage';
+
+import CustomRoute from './components/CustomRoute';
 
 const SignUp = lazy(() => import('~screens/SignUp'));
 const Login = lazy(() => import('~screens/Login'));
+const Home = lazy(() => import('~screens/Home'));
 
 function Routes() {
+  const token = loadStorage('accessToken');
+
   return (
     <BrowserRouter>
       <Switch>
-        <Route exact path={ROUTES.signUp} component={SignUp} />
-        <Route exact path={ROUTES.login} component={Login} />
-        <Redirect to={ROUTES.login} />
+        <Route
+          exact
+          path={ROUTES.base}
+          render={props => (token ? <Home {...props} /> : <Login {...props} />)}
+        />
```

Comment:
i would keep this for public routes, and i would make another router in the home screen using your customView to split responsibilities
```
<Switch>
  <CustomRoute exact path="/books" .......
```